### DISCUSSION TITLE
Character Training Advice?

### DISCUSSION THREAD

**Discussion by wktra** [TheLastBen/fast-stable-diffusion#481] [Q&A]
I'm training a very simplistic character I created myself (she is drawn in a 2D flat colored anime style). I drew and illustrated her in 50 different poses/angles/expressions. All of the images are very similarly proportional using the same design. This took me 3 weeks to painstakingly create. 

I want to train on an anime model, so that I can replicate my character in anime-art poses and situations that would take me endless hours to draw.  It'd be cool if I can generate this character in the style of Akira Toriyama or watercolor brushes or whatever, BUT my end goal is to simply be able to generate images that closely resemble my character. 

@TheLastBen what's your recommendation as far as training goes for this purpose? 200 steps per image? What about text encoding? 

Any advice is totally appreciated!

**Comment by TheLastBen** [comment]
If you want to replicate the exact character, set the text encoder to 100% and the steps to 4000, and if you're having issues applying style to it, redo (not resume) the training with 30% text_encoder and 6000 steps

**Reply by wktra** [reply]
@TheLastBen thank you so much!!!! >o<;
I started a training with text encoder set to 100% and the steps to 10,000 (because I had 50 images). 
Did I overdo it?

**Reply by TheLastBen** [reply]
limit the number of steps of the text encoder to 3000-4000, then you can continue training without it

**Reply by wktra** [reply]
@TheLastBen continue training without it? You mean setting the text encoder to "0"? 

And that's if I don't like the results, right?

**Reply by TheLastBen** [reply]
Setting the text_encoder to 100% for 10.000 steps will cause overfitting and impossible style transfer, so make sure you train it only for 3 or 4k steps

**Reply by wktra** [reply]
> limit the number of steps of the text encoder to 3000-4000, then you can continue training without it

Thank you so much for the advice 😭but could you please explain what you meant by the above? @TheLastBen 
I'm a noobette, sorry

What did you mean "continue training without it"? 
What do you mean by "it"?

**Reply by TheLastBen** [reply]
it, meaning the text_encoder, just uncheck the box "train_text_encoder" and resume training

**Reply by wktra** [reply]
Ok, so let's see if I understand this correctly:

**start training my 50 images at 4000-5000 steps with the text-encoder set to 100%. 
If I am not satisfied, then I can train for another 4000 steps _without_ the text encoding.** 

and if this doesn't work out at all:

**start training my 50 images at 6000 steps with the text encoder set to 30%.** 

Is this correct? (thank you in advance) @TheLastBen

**Reply by TheLastBen** [reply]
yes, sounds right

**Reply by wktra** [reply]
thank you so much 🤧

**Reply by TheLastBen** [reply]
happy to help

**Reply by wktra** [reply]
@TheLastBen Ok, so I ended up with two ckpts using your two suggestions:
a) 5000 steps, text encoding 100%
b) 6000 steps, text encoding 30%

both produce different results with settings like...euler-a, 50 steps, cfg 7.

Both are amazing and represent the character VERY well. 

Do you have any suggestions for tests I can run to determine which one is better? Or is it just subjective at this point?

**Reply by TheLastBen** [reply]
you'll have to experiment with different prompts and situations and styles to see which one complies the best

**Reply by wktra** [reply]
@TheLastBen as usual, your advice came through for me. >_< 
I'm curious: what are the benefits of training my illustrated character using images larger than 512px?

I originally drew my character poses and angles at 2000 pixels. Is it worth it to retrain using bigger images?

**Reply by TheLastBen** [reply]
Higher resolution means more information for the model to use, so yes, bigger resolution = better results

**Comment by wktra** [comment]
@TheLastBen 
Okay, so I have an update.

I trained 45 high quality pics of my illustrated girl character. 1024 x 1024. Used 4500 steps, 100% text-encoder. 8 hour training session. 
The results were extremely disappointing. I could not make it generate my character unless I specified aspects of her. And all of those generations were nowhere near close to my character style.

Gonna stick to 512x512 from now on.

**EDIT: OMG I'm am IDIOT. I used numbers in the name of the instance and image filename. The script stripped the numbers, so in the prompt I was calling up a nonexistent subject.**

**Reply by TheLastBen** [reply]
use only letters for the identifiers, the spaces and the underscores are stripped

**Reply by wktra** [reply]
@TheLastBen ok, new training finished just now. better instance and images name: "slpp" 
1024px images, 5000 steps, 100% encoder. 
Absolutely zero likeness to the character design on various settings. 
Trying again with 512px images

**Reply by TheLastBen** [reply]
"Absolutely zero likeness to the character design on various settings."

that means there is a problem with your settings or images
